### PR TITLE
Intro HashMap23::has, Timespec::millisec|microsec and misc clean up

### PR DESCRIPTION
**Pull request by xymus**

### CODE REVIEW

File: lib/pthreads/pthreads.nit [hunk 1]
```diff
@@ -155,7 +155,7 @@ end
 private extern class NativePthreadMutex in "C" `{ pthread_mutex_t * `}
 	new (attr: NativePthreadMutexAttr) `{
 		pthread_mutex_t *mutex = malloc(sizeof(pthread_mutex_t));
-		int res = pthread_mutex_init(mutex, attr);
+		pthread_mutex_init(mutex, attr);
```
Comment (privat) [inline lib/pthreads/pthreads.nit]: I'm not a fan of ignoring potential errors here.

File: lib/realtime.nit [hunk 2]
```diff
@@ -72,8 +72,19 @@ extern class Timespec `{struct timespec*`}
 		return self->tv_nsec;
 	`}
 
-	# Seconds in Float
-	# Loss of precision but great to print
+	# Elapsed time in microseconds, with both whole seconds and the rest
+	fun microsec: Int `{
```
Comment (privat) [inline lib/realtime.nit]: what about overflow?

**Comment by xymus** [comment]
Rerolled for @privat comments.

**Comment by privat** [comment]
+1

**Comment by R4PaSs** [comment]
:shipit:

**Pull request merged**